### ISSUE TITLE
supervisor v12.1.1 not working well with livepush

### ISSUE THREAD

**Issue by pdcastro** [balena-io/balena-supervisor#1512]
As originally reported by Shaun here:
https://www.flowdock.com/app/rulemotion/i-cli/threads/NDIXJP3vrEsNHSz57tuO9UneDNw

```
$ balena push 192.168.1.36 -s ~/balena-io-examples/balena-node-hello-world
$ /usr/local/bin/balena push 192.168.1.36 -s ~/balena/balena-io-examples/balena-node-hello-world
[Info]    Starting build on device 192.168.1.36
[Info]    No "docker-compose.yml" file found at "/Users/paulo/balena/balena-io-examples/balena-node-hello-world"
[Info]    Creating default composition with source: "/Users/paulo/balena/balena-io-examples/balena-node-hello-world"
[Build]   [main] Step 1/10 : FROM balenalib/raspberrypi3-node:10-stretch-run
[Build]   [main]  ---> 258e60f37899
[Build]   [main] Step 2/10 : WORKDIR /usr/src/app
[Build]   [main]  ---> Using cache
[Build]   [main]  ---> bfb39d98772a
[Build]   [main] Step 3/10 : COPY package.json package.json
[Build]   [main]  ---> Using cache
[Build]   [main]  ---> 0a7a4a9bca9d
[Build]   [main] Step 4/10 : RUN JOBS=MAX npm install --production --unsafe-perm && npm cache verify && rm -rf /tmp/*
[Build]   [main]  ---> Using cache
[Build]   [main]  ---> 226a41c0745d
[Build]   [main] Step 5/10 : COPY . ./
[Build]   [main]  ---> c7c658933754
[Build]   [main] Step 6/10 : ENV UDEV=1
[Build]   [main]  ---> Running in ca5a7bf2f0f5
[Build]   [main] Removing intermediate container ca5a7bf2f0f5
[Build]   [main]  ---> 9c31ef7a2176
[Build]   [main] Step 7/10 : ENV test=7
[Build]   [main]  ---> Running in 4a4b11ad0969
[Build]   [main] Removing intermediate container 4a4b11ad0969
[Build]   [main]  ---> 4f06c82b84db
[Build]   [main] Step 8/10 : CMD ["npm", "start"]
[Build]   [main]  ---> Running in d2b18cd715e4
[Build]   [main] Removing intermediate container d2b18cd715e4
[Build]   [main]  ---> 5af76ac9b26d
[Build]   [main] Step 9/10 : LABEL io.resin.local.image=1
[Build]   [main]  ---> Running in ae3aab977366
[Build]   [main] Removing intermediate container ae3aab977366
[Build]   [main]  ---> 36b26f19fa4c
[Build]   [main] Step 10/10 : LABEL io.resin.local.service=main
[Build]   [main]  ---> Running in 589650af1a2f
[Build]   [main] Removing intermediate container 589650af1a2f
[Build]   [main]  ---> 09dd52acc39a
[Build]   [main] Successfully built 09dd52acc39a
[Build]   [main] Successfully tagged local_image_main:latest

[Live]    Waiting for device state to settle...
[Info]    Streaming device logs...
[Live]    Watching for file changes...
[Logs]    [11/4/2020, 5:03:17 PM] Killing service 'main sha256:848427d2db36406ffa8672e3dc63eb4148f8b2643d81ad9423ca8b3c9fb71479'
[Logs]    [11/4/2020, 5:03:23 PM] Service exited 'main sha256:848427d2db36406ffa8672e3dc63eb4148f8b2643d81ad9423ca8b3c9fb71479'
[Logs]    [11/4/2020, 5:03:23 PM] Killed service 'main sha256:848427d2db36406ffa8672e3dc63eb4148f8b2643d81ad9423ca8b3c9fb71479'
[Logs]    [11/4/2020, 5:03:24 PM] Installing service 'main sha256:09dd52acc39a1594fa3bf93fbf0a6e7ffc89f57caef5763c2ad85aee0f9a4295'
[Logs]    [11/4/2020, 5:03:27 PM] Installed service 'main sha256:09dd52acc39a1594fa3bf93fbf0a6e7ffc89f57caef5763c2ad85aee0f9a4295'
[Logs]    [11/4/2020, 5:03:27 PM] Starting service 'main sha256:09dd52acc39a1594fa3bf93fbf0a6e7ffc89f57caef5763c2ad85aee0f9a4295'
[Logs]    [11/4/2020, 5:03:33 PM] Started service 'main sha256:09dd52acc39a1594fa3bf93fbf0a6e7ffc89f57caef5763c2ad85aee0f9a4295'
[Live]    Device state settled
[Logs]    [11/4/2020, 5:03:37 PM] [main]
[Logs]    [11/4/2020, 5:03:37 PM] [main] > simple-server-node@1.0.0 start /usr/src/app
[Logs]    [11/4/2020, 5:03:37 PM] [main] > node server.js
[Logs]    [11/4/2020, 5:03:37 PM] [main]
[Logs]    [11/4/2020, 5:03:39 PM] [main] Example app listening on port  80
```

Nothing wrong the output above, the problem is that when the project's Dockerfile is modified, live push will not automatically rebuild the app containers. In terms of console output, it is as if nothing is happening at all, as if the Dockerfile had not been modified.

My first suspicion was that the CLI was not detecting the Dockerfile changes, but the problematic behaviour does not happen with older supervisor versions, specifically does not happen with supervisor v11.14.0, but happens with supervisor version v12.1.1. I didn't have time to investigate it further. I am creating this issue sooner than later because @shaunmulligan pointed out: _"that supervisor version is included in meta-balena 2.61.0 and i assume the devices guys will start rolling that out next week."_

Tested with balenaOS v2.60.1+rev1.dev on an RPi 3, upgrading the supervisor from the "default" v11.14.0 to v12.1.1 using the JF KB entry [On device supervisor update for support agents](https://jel.ly.fish/88b558ac-dab9-42ea-a7e5-659d78ed68af).
Tested with multiple CLI versions - no difference between CLI versions. A reference CLI version is v12.26.1 (latest at the time of testing).

**Comment by CameronDiver** [comment]
This is interesting, as the supervisor version should not make a difference to livepush so getting to the bottom of this would be good.

**Comment by 20k-ultra** [comment]
When I was using livepush the CLI detected changes because it logs
```
[Debug] Got an add filesystem event for service: main. File: app/main.go
```

but then nothing happens.

CLI: 12.20.0
SV: 11.14.0
OS: balenaOS 2.58.6+rev1

**Comment by 20k-ultra** [comment]
The CLI watches your project for changes and what's interesting about that is when running the SV version that breaks livepush, the CLI does not log "Got an add filesystem event for service..." which is logged when there's a change in the file. So I'll do some digging in the CLI to see why.

**Comment by 20k-ultra** [comment]
I've figured out that because of the scoped api key change we made the CLI does not query the /v2/state/status endpoint with the correct authentication and so no containers are returned which makes the CLI return before setting up the file watchers in: https://github.com/balena-io/balena-cli/blob/20e0810d2a4d4cdbf4c92951b94724f006c587c2/lib/utils/device/live.ts#L136

I will be creating a patch to allow these types of requests to work like in the past if the device is in localmode. We will also add more test coverage since the `/v2/state/status` endpoint that broke this does not have sufficient testing. Ideally, the CLI will implement scoped api keys in the future regardless of the localmode check.

**Comment by richbayliss** [comment]
Should be resolved in #1516

**Comment by pdcastro** [comment]
As a summary for people still reporting similar livepush issues, this issue is understood to have affected supervisor versions v11.14.4 to v12.2.2, and thus balenaOS versions v2.61.0 to v2.64.4 (which ship with supervisor version v12.1.1, as per [balenaOS changelog](https://github.com/balena-os/meta-balena/blob/master/CHANGELOG.md)).

Besides #1516, another related supervisor PR is #1520.
